### PR TITLE
Closes #93 - improve the pipeline

### LINKED ISSUE #93: Split the "deploy artifacts to Sonatype" action, so that the "update Version to SNAPSHOT" step is extra
At the moment, if the "update Version to SNAPSHOT" step fails, then all parts of "deploy artifacts to Sonatype" need to be rerun. That's a problem, because in this case we need to delete the deployment from Sonatype

### PR DESCRIPTION
**Pull request by ryzheboka**
updating Kadai version to next Snapshot is now executed as an extra job instead of just being a step

### Thanks for your PR! Please fill out the following list :)

---

- [x] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [ ] Sonarcloud link : \<add the link here>
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [x] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -392,6 +392,16 @@ jobs:
           GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
           MAVEN_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
           MAVEN_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+      - name: Cancel workflow
+        if: failure()
+        uses: andymckay/cancel-action@0.5
```
Comment (CRoberto1926) [inline .github/workflows/continuous-integration.yml:397]: is this also going to revert the changes from change_version.sh?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -392,6 +392,16 @@ jobs:
...
+
+  update_version:
+    runs-on: ubuntu-22.04
+    name: Update version to next snapshot and push back
+    if: github.repository == 'kadai-io/kadai' && startsWith(github.ref, 'refs/tags/v') && github.head_ref == ''
+    needs: [ test_frontend, test_e2e, test_backend ]
```
Comment (CRoberto1926) [inline .github/workflows/continuous-integration.yml]: I would say needs: [release_artifacts] or? Would that work?

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -392,6 +392,16 @@ jobs:
           GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
           MAVEN_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
           MAVEN_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+      - name: Cancel workflow
+        if: failure()
+        uses: andymckay/cancel-action@0.5
+
+  update_version:
+    runs-on: ubuntu-22.04
```
Comment (CRoberto1926) [inline .github/workflows/continuous-integration.yml:400]: This is not part of your PR, but I would suggest to use everywhere the same ubuntu version. The last LTS should be 24.04, but we have to check if everything works. Could you write an issue for this please?

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (ryzheboka) [reply .github/workflows/continuous-integration.yml:397]: As change_version.sh doesn't commit anything or upload any artifacts, I see no reason why it wouldn't be reverted. How could we check this for sure?

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (ryzheboka) [reply .github/workflows/continuous-integration.yml:400]: Yes, I could introduce a constant for the Ubuntu version and set it to 22.04, as we know that it works. I can also create an issue for the update to 24.04

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (ryzheboka) [reply .github/workflows/continuous-integration.yml]: Yes :)

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (CRoberto1926) [reply .github/workflows/continuous-integration.yml:397]: mvn versions:commit is not committing anything?

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (CRoberto1926) [reply .github/workflows/continuous-integration.yml:400]: I would be in favor of a constant for ubuntu-22.04, yes

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (ryzheboka) [reply .github/workflows/continuous-integration.yml:397]: As can be seen in our repo history, after a release, we have one automatical commit, for example the "[Updated poms to version 9.1.1-SNAPSHOT](https://github.com/kadai-io/kadai/commit/24090bd351d365d9a2adc207776c7003e1dcba9f)" commit for last update. This commit is done in the next step in ci/commitPoms.sh. The step that I added the cacelation to was previously also there, but produced no additional commit. According to the [doc](https://www.mojohaus.org/versions/versions-maven-plugin/commit-mojo.html), mvn versions:commit doesn't commit anything in the classical sense. However, it still would be better to prove or try out if the changes by the change_version scripts are revertable by the cancel actions.

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (CRoberto1926) [reply .github/workflows/continuous-integration.yml]: Can you change that? Or do you have counter-arguments?

**Review by CRoberto1926 — APPROVED** [review]
